### PR TITLE
framework/ble_manager: refactor functions of 'bonded device'

### PR DESCRIPTION
**Pull request by an4967**
- integrate 'bonded device' control functions.
- add examples of 'bonded device' control functions.

### CODE REVIEW

File: framework/src/ble_manager/ble_manager_state.c [hunk 1]
```diff
@@ -120,31 +120,37 @@ ble_result_e blemgr_handle_request(blemgr_msg_s *msg)
...
-		trble_mode_e mode = *(trble_mode_e *)msg->param;
+		ret = ble_drv_delete_bonded_all();
+	} break;
 
-		ret = ble_drv_delete_bond_all(mode);
+	// Client
+	case BLE_EVT_CMD_SET_CALLBACK: {
+		BLE_STATE_CHECK;
+
+		memcpy(&g_client_config, msg->param, sizeof(ble_client_init_config));
```
Comment (jeongchanKim) [inline framework/src/ble_manager/ble_manager_state.c:150]: To use `memcpy`, please include `string.h`.

File: os/board/rtl8721csm/src/component/os/tizenrt/rtk_blemgr.c [hunk 2]
```diff
@@ -30,13 +30,29 @@
 
 extern unsigned char rltk_wlan_running(unsigned char idx);
 
+static void _reverse_mac(uint8_t *mac)
+{
+	int i;
+	int j;
+	uint8_t temp;
+
+	for (i = 0, j = TRBLE_BD_ADDR_MAX_LEN - 1;i < j; i++, j--) {
```
Comment (jeongchanKim) [inline os/board/rtl8721csm/src/component/os/tizenrt/rtk_blemgr.c:39]: Please add space after `;`.

File: os/board/rtl8721csm/src/component/os/tizenrt/rtk_blemgr.c [hunk 3]
```diff
@@ -30,13 +30,29 @@
...
+	int j;
+	uint8_t temp;
+
+	for (i = 0, j = TRBLE_BD_ADDR_MAX_LEN - 1;i < j; i++, j--) {
+		temp = mac[i];
+		mac[i] = mac[j];
+		mac[j] = temp;
+	}
+
+	return;
```
Comment (jeongchanKim) [inline os/board/rtl8721csm/src/component/os/tizenrt/rtk_blemgr.c:45]: I think this line is not needed.

File: os/net/blemgr/bledev.c [hunk 4]
```diff
@@ -84,32 +84,40 @@ int bledev_handle(struct bledev *dev, lwnl_req cmd, void *data, uint32_t data_le
...
 		} else {
 			return TRBLE_INVALID_ARGS;
 		}
-		addr = (trble_bd_addr *)param.param[0];
-		mode = *(trble_mode_e *)param.param[1];
+		device_list = (trble_bonded_device_list_s *)param.param[0];
+		device_count = (uint16_t *)param.param[1];
 
-		TRBLE_DRV_CALL(ret, dev, del_bond, (dev, addr, mode));
+		TRBLE_DRV_CALL(ret, dev, get_bonded_dev, (dev, device_list, device_count));
```
Comment (jeongchanKim) [inline os/net/blemgr/bledev.c:101]: `device_count` is pointer type. Most of other `TRBLE_DRV_CALL`, it passes the int variable.
Is it okay to pass the pointer type?

File: framework/src/ble_manager/ble_manager_state.c (hunk 1 above)
Comment (an4967) [reply framework/src/ble_manager/ble_manager_state.c:146]: Okay, I added it

File: os/board/rtl8721csm/src/component/os/tizenrt/rtk_blemgr.c (hunk 2 above)
Comment (an4967) [reply os/board/rtl8721csm/src/component/os/tizenrt/rtk_blemgr.c]: Okay, I fixed it

File: os/board/rtl8721csm/src/component/os/tizenrt/rtk_blemgr.c (hunk 3 above)
Comment (an4967) [reply os/board/rtl8721csm/src/component/os/tizenrt/rtk_blemgr.c]: Okay, I removed it

File: os/net/blemgr/bledev.c (hunk 4 above)
Comment (an4967) [reply os/net/blemgr/bledev.c:101]: 'device_count' is for getting the number of bonded devices.  To do that, this should be pointer

**Review by pillip8282 — APPROVED** [review]
lgtm

**Pull request merged**